### PR TITLE
added verifySideNav

### PR DESCRIPTION
**Pull request by tbeck860**
1st PR - Adds automated testing for the side nav routes and UI

(builds off of the work started in the `feature/end-to-end-tests` branch)

### CODE REVIEW

**Comment by comountainclimber** [comment]
This is awesome @tbeck860 ! A few things can you be sure to add assertions here in addition to running spectron (like the ones that exist in `view-balance.e2e.js`)

I am also wondering if we should encapsulate all of the tests that involved being authenticated in a single file rather than duplicating the code to login

**Comment by tbeck860** [comment]
> This is awesome @tbeck860 ! A few things can you be sure to add assertions here in addition to running spectron and these automated clicks (like the ones that exist in `view-balance.e2e.js`)
> 
> I am also wondering if we should encapsulate all of the tests that involved being authenticated in a single file rather than duplicating the code to login. Let me know your thoughts

@comountainclimber If waitUntillTextExists fails the test will fail but the error is undefined. I am still trying to catch this error (promise<void) and output something more meaningful.

File: package.json [hunk 1]
```diff
@@ -129,7 +129,8 @@
     "sifter": "0.5.3",
     "spunky": "1.3.1",
     "svg-react-loader": "0.4.5",
-    "uuid": "3.2.1"
+    "uuid": "3.2.1",
+    "webdriverio": "^5.8.4"
```
Comment (comountainclimber) [inline package.json]: Why exactly was this added? I dont see it being used in the new test

File: __e2e__/verifySideNav.e2e.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ test.after(async () => {
   }
 })
 
-test.serial('should login successfully', async t => {
+test.serial('should login successfully', async e => {
```
Comment (comountainclimber) [inline __e2e__/verifySideNav.e2e.js]: I prefer that this variable be kept `t` as it is the ava the test instance (not an error)

File: __e2e__/verifySideNav.e2e.js [hunk 3]
```diff
@@ -43,41 +43,56 @@ test.serial('should login successfully', async t => {
...
-
+  await app.client.waitUntilTextExists('//h3', 'Settings')
+  e.pass()
+})
+test.serial('Logout button clicked', async e => {
   //click logout
-  //await app.client.click('#logout')
-  //await app.client.waitUntilWindowLoaded(10000)
+  await app.client.click('#logout')
+  await app.client.waitUntilTextExists('//div[contains(@class,\'_3M2L1lj7oJRu_0KBN3_v87\')]', 'Login')
```
Comment (comountainclimber) [inline __e2e__/verifySideNav.e2e.js]: I am worried about checks on dynamically generated class name hashes like this one... Can we just add an `id` tag to the element you are checking instead?

**Comment by comountainclimber** [comment]
Keep up the good work @tbeck860 I have some feedback for you based on the latest changes

**Comment by comountainclimber** [comment]
@tbeck860 this looks like its ready to go! Nice job... Last bit of feedback and then we can merge: Can you revert your changes to both `package.json` and `package-lock.json` - those _should_ remain unchanged with this PR

**Comment by comountainclimber** [comment]
Please award @tbeck860 200 points

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/c382c87ce7282664dc849fef5cb8c8de22cf3b596dd1b4e85c87efb194bb2003

**Pull request merged**